File: src/main/webapp/src/app/models/activity-audio-bus.ts
import { ActivityAudioChannel } from "./activity-audio-channel";

export class ActivityAudioBus {
    name: string;
    activityAudioChannelList: ActivityAudioChannel[] = [];

    constructor(data?: any) {
        if(!data) {
        	return;
        }

        this.name = data.name;

        this.activityAudioChannelList = [];

        for(let activityAudioBus of data.activityAudioChannelList) {
            this.activityAudioChannelList.push(new ActivityAudioChannel(activityAudioBus));
        }
    }
}

File: src/main/java/com/ascargon/rocketshow/api/ActivityAudioBus.java
package com.ascargon.rocketshow.api;

import javax.xml.bind.annotation.XmlRootElement;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement
public class ActivityAudioBus {

    private String name;

    private List<ActivityAudioChannel> activityAudioChannelList = new ArrayList<>();

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<ActivityAudioChannel> getActivityAudioChannelList() {
        return activityAudioChannelList;
    }

    public void setActivityAudioChannelList(List<ActivityAudioChannel> activityAudioChannelList) {
        this.activityAudioChannelList = activityAudioChannelList;
    }

}
